openwillis/measures/video/eye_blink.py

```diff
@@ -0,0 +1,285 @@
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs, properties = find_peaks(-framewise['EAR'], prominence=prominence, width=width)
+    left_ips = properties["left_ips"]
+    right_ips = properties["right_ips"]
+    # round to nearest integer and add 1 to match frame number
+    left_ips = np.round(left_ips).astype(int) + 1
+    right_ips = np.round(right_ips).astype(int) + 1
+    troughs = np.round(troughs).astype(int) + 1
+    return troughs, left_ips, right_ips
+
+
+def convert_frame_to_time(troughs, left_ips, right_ips, fps):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function converts the frame number to time.
+
+    Parameters:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+    fps : float
+        The fps of the video
+
+    Returns:
+    ............
+    blinks : pd.DataFrame
+        Contains for each blink the frame number and the time (in seconds)
+         start of the blink and end of the blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs_time = troughs/fps
+    left_ips_time = left_ips/fps
+    right_ips_time = right_ips/fps
+
+    blinks = pd.DataFrame({'Blink Peak Frame': troughs, 'Blink Starting Frame': left_ips, 'Blink Ending Frame': right_ips,
+                            'Blink Peak Time': troughs_time, 'Blink Starting Time': left_ips_time, 'Blink Ending Time': right_ips_time})
+    return blinks
+
+
+def eye_blink_rate(video_directory):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function counts the number of eye blinks in a given video.
+
+    Parameters:
+    ............
+    video_directory : string
+        The directory of the video to be analyzed
+
+    Returns:
+    ............
+    framewise: pd.DataFrame
+        Contains the frame number and the eye aspect ratio (EAR) of each frame
+    blinks : pd.DataFrame
+        Contains for each blink the frame number and the time (in seconds)
+         start of the blink and end of the blink
+    summary : pd.DataFrame
+        The number of eye blinks and blink rate (blinks per minute)
+
+    Raises:
+    ............
+    ValueError
+        If device is not 'laptop' or 'mobile'
+
+    ---------------------------------------------------------------------------------------------------
+    """
+
+    framewise, blinks, summary = None, None, None
+
+    try:
```

Comment:
Just a suggestion: Add exception handling for specific exceptions rather than using a general catch-all

Reply:
I can change it to only _Value_Error_, since this is the only error I expect to happen, but I am unsure if that is the only error that can occur. I could split the responses to one for _Value_Error_ and one for unexpected behaviors, _Exception_, but I simply log the error either way, so I am not sure we need to make more specific _except_ statements for different errors.